scrapy/selector/unified.py

```diff
@@ -47,113 +26,37 @@ def _response_from_text(text, st):
               body=to_bytes(text, 'utf-8'))
 
 
-class Selector(object_ref):
-
-    __slots__ = ['response', 'text', 'namespaces', 'type', '_expr', '_root',
-                 '__weakref__', '_parser', '_csstranslator', '_tostring_method']
+class Selector(ParselSelector, object_ref):
 
-    _default_type = None
-    _default_namespaces = {
-        "re": "http://exslt.org/regular-expressions",
+    __slots__ = ['response']
 
-        # supported in libxslt:
-        # set:difference
-        # set:has-same-node
-        # set:intersection
-        # set:leading
-        # set:trailing
-        "set": "http://exslt.org/sets"
-    }
-    _lxml_smart_strings = False
+    def __init__(self, response=None, text=None, type=None, root=None, _root=None, **kwargs):
+        st = _st(response, type or self._default_type)
 
-    def __init__(self, response=None, text=None, type=None, namespaces=None,
-                 _root=None, _expr=None):
-        self.type = st = _st(response, type or self._default_type)
-        self._parser = _ctgroup[st]['_parser']
-        self._csstranslator = _ctgroup[st]['_csstranslator']
-        self._tostring_method = _ctgroup[st]['_tostring_method']
+        if _root is not None:
+            warnings.warn("Argument `_root` is deprecated, use `root` instead",
+                          ScrapyDeprecationWarning, stacklevel=2)
+            if root is None:
+                root = _root
+            else:
+                warnings.warn("Ignoring deprecated `_root` argument, using provided `root`")
 
         if text is not None:
             response = _response_from_text(text, st)
 
         if response is not None:
-            _root = LxmlDocument(response, self._parser)
+            text = response.body_as_unicode()
```

Comment:
now that parsel can handle base_url, let's be full backward compatible: `kwargs.setdefault('base_url', response.url)`